apps/adoption/src/app/components/demo.component.ts

```diff
@@ -0,0 +1,182 @@
+              <cds-icon shape="terminal"></cds-icon> Run localy
+            </button>
+            <button (click)="openCodesandbox(tab)" class="btn btn-sm btn-link">
+              <cds-icon shape="code"></cds-icon> Codesandbox
+              <cds-icon size="md" shape="beta" status="success"></cds-icon>
+            </button>
+          </clr-tab-content>
+        </clr-tab>
+
+        <clr-tab *ngIf="!sideBySideState">
+          <button clrTabLink (click)="sideBySide()" [clrTabLinkInOverflow]="inOverflow">
+            <cds-icon (click)="sideBySide()" size="md" shape="view-columns">Side by side</cds-icon> Side by side
+          </button>
+        </clr-tab>
+
+        <clr-tab *ngIf="sideBySideState">
+          <button clrTabLink (click)="sideBySide()" [clrTabLinkInOverflow]="inOverflow">
+            <cds-icon size="md" shape="landscape"></cds-icon> Full width
+          </button>
+        </clr-tab>
+      </clr-tabs>
+    </div>
+    <div *ngIf="!sideBySideState">
+      <clr-tabs>
+        <clr-tab *ngFor="let tab of tabs">
+          <button clrTabLink>{{ tab.name }}</button>
+          <clr-tab-content *clrIfActive>
+            <div *ngFor="let file of tab.files | keyvalue">
+              <sourcecode [src]="file.value" [language]="tab.language"></sourcecode>
+            </div>
+            <div id="{{ tab.id }}"></div>
+            <button (click)="openStackblitz(tab)" class="btn btn-sm btn-link">
+              <cds-icon shape="bolt"></cds-icon> StackBlitz
+            </button>
+            <button (click)="embedStackblitz(tab, tab.id)" class="btn btn-sm btn-link">
+              <cds-icon shape="terminal"></cds-icon> Run
+            </button>
+            <button (click)="openCodesandbox(tab)" class="btn btn-sm btn-link">
+              <cds-icon shape="code"></cds-icon> Codesandbox
+              <cds-icon size="md" shape="beta" status="success"></cds-icon>
+            </button>
+          </clr-tab-content>
+        </clr-tab>
+
+        <clr-tab *ngIf="!sideBySideState">
+          <button clrTabLink (click)="sideBySide()" [clrTabLinkInOverflow]="inOverflow">
+            <cds-icon (click)="sideBySide()" size="md" shape="view-columns">Side by side</cds-icon> Side by side
+          </button>
+        </clr-tab>
+
+        <clr-tab *ngIf="sideBySideState">
+          <button clrTabLink (click)="sideBySide()" [clrTabLinkInOverflow]="inOverflow">
+            <cds-icon size="md" shape="landscape"></cds-icon> Full width
+          </button>
+        </clr-tab>
+      </clr-tabs>
+    </div>
+  `,
+  providers: [StackblitzService, CodesandboxService],
+})
+export class Demo {
+  inOverflow = true;
+
+  @Input('tabs') tabs: DemoTabData[] = [];
+
+  // Experimental
+  @Input('sideBySide') _sideBySideState: boolean | string = false;
+
+  constructor(private stackblitz: StackblitzService, private codesandbox: CodesandboxService) {}
+
+  ngOnInit(): void {
+    if (this.tabs) {
```

Comment:
```suggestion
    if (this.tabs && this.tabs.length) {
```

Reply:
Basically this is handle below but a better check may be will be 
```
if (Array.isArray(this.tabs)) { 
```

`this.tabs.length` won't prevent `String` from passing